lib/bitvec.cpp

```diff
@@ -36,6 +36,7 @@ std::ostream &operator<<(std::ostream &os, const bitvec &bv) {
 }
 
 bitvec &bitvec::operator>>=(size_t count) {
+    assert(static_cast<int>(count) >= 0);
```

Comment:
The C++ spec says that operator new cannot ever return NULL (unlike malloc) -- it will throw a bad_alloc exception instead.  If we're seeing NULLs, that is a bug (probably in the garbage collector).